src/Test/Perf/util/test_util.csx

```diff
@@ -12,125 +12,187 @@ using System.Threading;
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Net;
+using System.Globalization;
 
-//
-// Arguments
-//
-
-// This is due to a design decision in csi that has Args non-static.
-// Non-static variables are impossible to read inide of nested classes.
-static IEnumerable<string> StaticArgs = null;
-StaticArgs = Args;
-
-/// Returns the path to log file if one exists.
-/// Returns null otherwise.
-static string LogFile()
+class RelativeDirectory 
 {
-    var key = "--log=";
-    return (from arg in StaticArgs where arg.StartsWith(key) select arg.Substring(key.Length)).FirstOrDefault();
-}
+    string _workingFile;
+    
+    public RelativeDirectory([CallerFilePath] string workingFile = "") 
+    {
+        _workingFile = workingFile;
+    }
+    
+    public string MyWorkingDirectory => Directory.GetParent(_workingFile).FullName;
```

Comment:
Everytime we call `MyWorkingDirectory`, this value is computed. Instead we can make MyWorkingDirectory a readonly property and the value can be assigned inside the `ctor` os `RelativeDirectory`